lib/screens/settings/audio_sources.dart

```diff
@@ -141,42 +143,122 @@ class _AudioSourcesScreenState extends State<AudioSourcesScreen> {
+                                                        Colors.white;
+
+                                                    Icon? icon;
+
+                                                    switch (value) {
+                                                      case TorchState.on:
+                                                        icon = const Icon(
+                                                          Icons.flash_off,
+                                                          color: iconColor,
+                                                        );
+                                                        break;
+                                                      case TorchState.off:
+                                                        icon = const Icon(
+                                                          Icons.flash_on,
+                                                          color: iconColor,
+                                                        );
+                                                        break;
+                                                    }
+
+                                                    return Row(
+                                                      mainAxisSize:
+                                                          MainAxisSize.max,
+                                                      mainAxisAlignment:
+                                                          MainAxisAlignment
+                                                              .spaceBetween,
+                                                      children: [
+                                                        IconButton(
+                                                          onPressed: () =>
+                                                              Navigator.pop(
+                                                                  ctx),
+                                                          icon: const Icon(
+                                                            Icons.close,
+                                                            color: iconColor,
+                                                          ),
+                                                        ),
+                                                        IconButton(
+                                                            onPressed: () =>
+                                                                _scanController
+                                                                    .toggleTorch(),
+                                                            icon: icon),
+                                                      ],
+                                                    );
+                                                  },
+                                                ),
+                                              ),
```

Comment:
Move this to a separate widget so it can be shared between the two scanner implementations.

Reply:
Ok this makes more sense now.